source/Applications/Advanced/CaptureUndistortRGB/CaptureUndistortRGB.cpp

```diff
@@ -187,6 +187,10 @@ int main()
     catch(const std::exception &e)
     {
         std::cerr << "Error: " << Zivid::toString(e) << std::endl;
-        return EXIT_FAILURE;
+        std::cout << "Press enter to exit." << std::endl;
+        if(std::cin.get() == '\n')
+        {
+            return EXIT_FAILURE;
+        }
```

Comment:
If you double click on the exe file on Windows the terminal will close and the user will not see the message. This will be perceived as though the program doesn't work.
This will not happen if you start the program from terminal, but most people don't start programs by opening the terminal but by double clicking on the exe file.

Reply:
With how it is stated in the tutorials the samples should be able to run with just access the release folder and double-clicking on the generated .exe. With that mindset, I thought adding in the pause for just printing errors might be smart.

Reply:
This gives quite some duplication in all samples? What convention do you have for avoiding duplication within these samples?

It could be put in a function that is used by all samples. Don't know if this might interfere with each sample being standalone? @SatjaSivcev

Reply:
We don't want to have complicated code. Each sample should be standalone and simple. Samples are not SDK so there is no need to use one function (file) for something like this by all samples. I don't think this as of now is too much code.
The intention of the samples is for the customers to learn how to use the SDK functions. We don't want them to spend time figuring out what the code does.

Reply:
> We don't want to have complicated code. Each sample should be standalone and simple. Samples are not SDK so there is no need to use one function (file) for something like this by all samples. I don't think this as of now is too much code.

I don't agree that it is more complicated. But definetly agree that it's a benefit that the sample is standalone (copy-paste-able); so if we want this user input on error I think we can have it like this 👍

Reply:
I agree this is weird/unstandard behavior and only complicated the source code, any CLI tool behaves this way if you double click it in the file explorer, that is just user error. I think it is better to update the documentation.

Reply:
I assume KB uses VS as an example for how to build on Windows? Then just say press "Ctrl+F5" to run the sample. (Ref: https://stackoverflow.com/a/1775870)

If KB documents show how to build on the console on Windows then just say "./build/path/sample.exe"

Reply:
You assume that every user (including those who double click by default) will read the documentation. I don't think they will. So for those who won't, the program will seem not to work and they will not see the error message (because they will double click the exe file).

Can you explain what is the benefit of the previous solution, for developers? I don't get it, except that it has 4 lines of code less? I can understand that it is standard use but imp it has limitations.

Reply:
IRL update: I think we mostly disagree because we have a different mental model of the end-user.

To make decisions about whether to do things like this in code and whether to spell out every tedious detail in KB I think we need two things.

1. One to three reference end-users with a particular skill set documented in Confluence that we optimize documentation experience for. These users can be used to align all documentation writers to the same mental model. and be referred to when having a dispute. Maybe also a couple of customer we do not optimize for (the expert and the one that cannot program at all, catering to those are just out of scope, we do not even try).
2. Clear rules about what is our domain to teach and what we can expect customers to find documentation for elsewhere. Example of guidelines:
    - We do not teach people how to program
    - We do not teach people how to debug
    - We do not teach people how to compile and run software in other tools than CMake and Visual Studio
    - We teach how to best integrate Zivid SW with 3rd party. For example Eigen, but we assume the customer knows Eigen as a standalone library, we do not teach Eigen. Then the customer should look up Egen doc on his own.
    - We do not teach best practices in C++ (We do not write an article in KB about RAII), but our examples are written using best practice

Having those two guidelines could help to make documentation more consistent regardless of author and it could help making decisions like this. 

FYI: @ajlandau, @torbsorb, @SatjaSivcev, @runenordmo, @dragly, @Martiix: What do you think about this?

Reply:
> Can you explain what is the benefit of the previous solution, for developers? I don't get it, except that it has 4 lines of code less?

I am not concerned about those four lines in particular, but I am concerned if our default is to inflate documentation by 30% to cater for a marginal or made up customer that will never be able to integrate Zivid into anything at all. All that extra nice is paid for by more experienced developers as mental overhead. My concern is that we optimize for the wrong end-user or that we try to cater to all which is just going to be bad for all.

Reply:
> Samples should be simple. "Look here only 20 lines to get going" and noise like this should be minimized imo.

I completely agree on this, and this was why I added the original comment.

> Having those two guidelines could help to make documentation more consistent regardless of author and it could help making decisions like this.

I definitely think that such guidelines could be helpful, both for writing and reviewing samples and documentation; as this is an example presenting potential misalignment between for instance the software and application team.

Reply:
I completely understand the purpose of those lines since I also am someone who has double clicked plenty of .exes that immediately closed in sample and bin folders of random software on Windows :facepalm:

I would, however, argue for removing those lines.

Defining the target user is definitely at the core of answering this and similar questions. And defining the target user also involves *choosing* a target user. That choice should be driven by value for our company. For instance, targeting experienced software engineers is likely more valuable than targeting first-year students. 

Now, I am not saying experienced software engineers will not double-click these files (I do it myself), but I assume that most *target users for the samples* are experienced software engineers who will conclude that they clicked a CLI program that needs to opened from CMD. That is, I think the intersection between people who do not understand the problem and are also target users are very few. This might be a wrong assumption, however, so please feel free to correct me on this.

However, if there are real and important users that are dissuaded by these samples not staying opened, perhaps you could you say more about who they are? It sounds like they could be equally important users (and decision makers) who would be much more happy if they found some nice GUI examples when double-clicking these .exes. In that case, we could perhaps divide our samples folder into a root folder with GUI examples and a sub-folder clearly marked as CLI samples?

All in all, I am not too bothered by these lines, but I would keep them out if it was up to me. It sounds like they are the wrong solution for the user who has the CMD closing-problem.

Reply:
I agree that we should set who we expect to be using the samples in the KB somewhere @nedrebo. My mindset when I was adding this is I was following the tutorials and realized with how my first impression is of the samples, that if I have a sample, I would expect it to work with just a double click, and if it doesn't, that it would tell me why. I did not see the few extra lines as a deterrent and if someone does see that, then wouldn't they already not enjoy the fact it is even in a try-catch loop?

If we say that the first person to interact with the camera is an engineer who is just buying one to investigate if they think it's good enough, at least in my experience it is not the guarantee that the person who will be integrating the camera will be the same person to first evaluate it. It is also not a guarantee that this person will be as immediately savvy as one would hope, but probably would have the engineering experience or seniority to still make a call over if it was worth giving to the people that do have the knowledge to integrate it into their systems.

So that boils down to for me is: 
-what do we expect users to gain from these samples?
-Are they for learning how our code works?
- are they for showing off how our camera works in their settings and without the GUI?
-Should we say that the Zivid studio interface is enough of a tool for those who are not SE to show off the camera features?